src/dc_imex.rs

```diff
@@ -278,9 +272,10 @@ fn set_self_key(
         .and_then(|(k, h)| if k.verify() { Some((k, h)) } else { None })
         .and_then(|(k, h)| k.split_key().map(|pub_key| (k, pub_key, h)));
 
-    if keys.is_none() {
-        bail!("File does not contain a valid private key.");
-    }
+    ensure!(
+        !keys.is_none(),
```

Comment:
please use  `keys.is_some()` instead of double negation :)

Reply:
done